File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    cnt=-1
    alpha = ["A","E","I","O","U"]
    
    def dfs(curr_word):
        nonlocal answer, cnt
        
        cnt+=1
        
        if curr_word ==word:
            answer= cnt
            return
        if len(curr_word)==5:
            return
        for a in alpha:
            if answer==0:
                dfs(curr_word+a)
        
    dfs("")
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;
int cnt=0;
vector<char> w = {'A', 'E', 'I', 'O', 'U'};
bool found = false;
int answer = 0;

void dfs(string& word, string curr){
    if(found) return;
    
    if(curr.size()==5){
        return;
    }
    for(auto a:w){
        string next = curr+a;
        cnt++;
        if(next==word){
            answer=cnt;
            found=true;
            return;
        }
         dfs(word, next);
    }
}


int solution(string word) {
    dfs(word, "");
    return answer;
}